app/application/use_cases/reset_password.py

```
@@ -0,0 +1,75 @@
+        repo: UserRepository,
+        reset_svc: PasswordResetService,
+        hasher: PasswordHasher,
+    ):
+        self._repo = repo
+        self._reset_svc = reset_svc
+        self._hasher = hasher
+
+    async def execute(self, inp: ConfirmPasswordResetInput) -> None:
+        email = await self._reset_svc.consume_token(inp.token)
+        if email is None:
+            raise ValueError("Invalid or expired reset token")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)